addons/mail/static/src/models/chat_window.js

```diff
@@ -158,6 +158,36 @@ registerModel({
                 this.update({ isFocused: true });
             }
         },
+        /**
+         * Save the scroll positions of the chat window in the store.
+         * This is useful in order to remount chat windows and keep previous
+         * scroll positions. This is necessary because when toggling on/off
+         * home menu, the chat windows have to be remade from scratch.
+         */
+        saveThreadScrollTop() {
+            if (
+                !this.threadView ||
+                !this.threadView.messageListView ||
+                !this.threadView.messageListView.component ||
+                !this.threadViewer
+            ) {
+                return;
+            }
+            if (
+                this.threadViewer.threadView &&
+                this.threadViewer.threadView.componentHintList.length > 0
+            ) {
+                // the current scroll position is likely incorrect due to the
+                // presence of hints to adjust it
+                return;
+            }
```

Comment:
I don't understand why we have 2 separate `if` to preemptively return this function; Other than the comment.